src/Commands/Self/ConsoleCommand.php

```diff
@@ -0,0 +1,32 @@
+<?php
+
+namespace Pantheon\Terminus\Commands\Self;
+
+use Pantheon\Terminus\Site\SiteAwareInterface;
+use Pantheon\Terminus\Site\SiteAwareTrait;
+use Pantheon\Terminus\Commands\TerminusCommand;
+
+class ConsoleCommand extends TerminusCommand implements SiteAwareInterface
+{
+    use SiteAwareTrait;
+    /**
+     * Initiates an interactive PHP console
+     *
+     * @command self:console
+     *
+     * @option string $site_env Site & environment to access as `$site` and (optional) `$env`
+     *
+     * @usage terminus self:console
+     *   Initiate an an interactive PHP console
+     * @usage terminus self:console mysite
+     *   Initiate an interactive PHP console with access to `$site` object
+     * @usage terminus self:console mysite.dev
+     *   Initiate an interactive PHP console with access to `$site` and `$env` objects
+     *
+     */
+    public function console($site_env = null)
+    {
+        list($site, $env) = $this->getOptionalSiteEnv($site_env, null);
```

Comment:
The PsySH shell creates an interactive console within context/scope of the place it's invoked. $site and $env have been created as variables within that context/scope, so they're available.